infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutTrackNumberService.kt

```diff
@@ -105,6 +106,7 @@ class LayoutTrackNumberService(
     fun getKmLengths(
         publishType: PublishType,
         trackNumberId: IntId<TrackLayoutTrackNumber>,
+        returnEmptyListOnInvalidGeocodingContext: Boolean = false,
```

Comment:
Ehkäpä vähän hämärä muutos funktion kannalta, mutta toisaalta hyvin yksinkertainen muutos tämän datasetin lisäämisen osalta. Nyt siis tätä samaa funktiota voidaan käyttää myös tilanteessa, jossa geokoodauskontekstin luominen ei onnistu (vaikkapa kun kilometrejä ei tietyllä ratanumerolla ole).

Reply:
Menisin sanomaan ehkä vähän turhankin hämäräksi. Ennemmin poistaisin tuon palautusarvon null-tarkistuksen kokonaan, ja sallisin vaan tämän funktion epäonnistua: Koska ainoa mahdollinen tapa epäonnistua on, että geokoodauskontekstia ei pystytty luomaan, riittää tätä yhtä tapausta edustamaan yksi palautustyypin arvo, eli miksei vaikka null.

Parin kutsujan pitää sitten erikseen päättää, mitä he haluavat tehdä siinä tapauksessa, että kutsu epäonnistui, mutta se on kutsujien asia päättää.

Reply:
Mjoo kyllähän se näin on järkevämpää, muuttelin tuon palauttamaan nullin.

---

Käyttöpaikkojen tutkimisen perusteella muutin aiemmat kutsujat palauttamaan tyhjät listat, jos ne saavat tuolta funkkarilta nyt nullin.

Aiemmin toinen näistä virheitä heittävän rajapinnan käyttäjistä oli
https://github.com/finnishtransportagency/geoviite/blob/df2a4d7592afbe5d2b59075e4e342bd8e46105bb/ui/src/track-layout/layout-km-post-api.ts#L174-L187

joka siis palautti jo vastaavasti (ui-koodin puolella) aiemmin virheellisen rajapintavastauksen jälkeen tyhjän listan omalle kutsujalleen, joten tyhjän listan palauttaminen myös bäkkäriltä "ok" arvona lienee soveltuva. Sanoisin epätodennäköiseksi, että tähän liittyvän rajapinnan HTTP-virhettä on käytetty jossain (ulkopuolisessa) yhteydessä tulkintana.

---

Vastaavasti toinen aiempi kutsuja
https://github.com/finnishtransportagency/geoviite/blob/df2a4d7592afbe5d2b59075e4e342bd8e46105bb/infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutTrackNumberService.kt#L150-L155
nakkasi tietyn ratanumeron CSV-pyynnöstä HTTP 500, jos km-pituuksia ei ratanumerolla ollut. Käyttäjä ei ole kuitenkaan voinut ui:n kautta suoraan yrittää tällaista CSV-tiedostoa hakea, sillä CSV-latausnappi km-pituuksittomille ratanumeroille on disabloitu. On siis melko epätodennäköistä, että tätäkään HTTP 500 virhettä on aiemmin jossain hyödynnetty. Muutettuna logiikkana siis km-pituuksittomalle ratanumerolle palautetaan CSV-tiedosto pelkällä otsikkorivillä, joka toisaalta myös vastaa käyttöliittymän näkymää.